### PR TITLE
net: socket: DNS API proposal

### PR DESCRIPTION
**Pull request by lemrey**
Two proposals for DNS API, any comments?

FYI: @metc @stig-bjorlykke

### CODE REVIEW

File: proposal.h [hunk 1]
```diff
@@ -0,0 +1,14 @@
+/* one */
+proto:
+int nrf_setdnsaddr(const char *dns, size_t len);
```
Comment (rlubos) [inline proposal.h]: How would you detect address family in this case?

Comment (rlubos) [reply proposal.h]: And if you specify in the API that you expect a string here, there's no need for a `len` field.

Comment (lemrey) [inline proposal.h]: or `nrf_setdnssrv` ?

File: proposal.h [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+usage:
+nrf_setdnsaddr("8.8.8.8", strlen("8.8.8.8)"));
+
+/* two */
+proto:
+int nrf_setdnsaddr(int family, void *in_addr);
```
Comment (rlubos) [inline proposal.h]: I was thinking about using `sockaddr` pointer, which has the `family` information within, and can hold both, IPv4 and IPv6. But I see that there's a binary incompability betweent Zephyr/nRF sockets, so perhaps it was not a good idea.

File: proposal.h [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+usage:
+struct nrf_in_addr addr;
+nrf_inet_pton(AF_INET, "8.8.8.8", &addr);
```
Comment (rlubos) [inline proposal.h]: so if we have to use `nrf_inet_pton` anyway, I don't see a point of doing this outside of the function. Theoretically, we could use `net_pton` and cast from in_addr/in6_addr to nrf_in_addr/nrf_in6_addr (these are binary compatible), but I'm not convinced it's a neat solution. What would you say?

**Comment by lemrey** [comment]
Thanks

**Pull request closed without merging**